File: angular-app/src/app/shared/models/address.ts
import {AddressDomain} from "./address.domain";

export class AddressDomainModel implements AddressDomain {
  public street: string;
  public city: string;
  public state: string;
  public zip: string;
  public county: string;

constructor(config: AddressDomain) {
  this.street = (config !== null && config !== undefined) ? config.street : undefined;
  this.city = (config !== null && config !== undefined) ? config.city : undefined;
  this.state = (config !== null && config !== undefined) ? config.state : undefined;
  this.zip = (config !== null && config !== undefined) ? config.zip : undefined;
  this.county = (config !== null && config !== undefined) ? config.county : undefined;
}
}

File: bp-web-ws/src/main/java/software/bytepushers/bpweb/model/entity/Address.java
package software.bytepushers.bpweb.model.entity;

import lombok.Getter;
import lombok.Setter;
import software.bytepushers.bpweb.model.entity.ValidatorModel.CreateRequest;

import javax.persistence.Column;
import javax.persistence.ElementCollection;
import javax.persistence.Entity;
import javax.validation.constraints.NotEmpty;
import java.util.List;

/**
 * The Address entity which would embedded to some other entity.
 */
@Getter
@Setter
@Entity(name = "T_ADDRESS")
public class Address extends AbstractEntity {

    @ElementCollection
    private List<String> street;

    @Column
    @NotEmpty(groups = CreateRequest.class, message = "{person.address.city.required}")
    private String city;

    @Column
    @NotEmpty(groups = CreateRequest.class, message = "{person.address.state.required}")
    private String state;

    @Column
    private String zip;

    @Column
    private String country;

    @Column
    private String county;
}
